### ISSUE TITLE
Add desktop support macOS

### ISSUE THREAD

**Issue by sebastianhaberey** [Nimrodda/flutter_nsd#11]
Great, this is exactly what I was looking for 👍 The code seems short and succinct, so that's good. Do you have any plans to support deskop platforms?

**Comment by Nimrodda** [comment]
Not needed for my current project, but if you feel like contributing it,
I'd be very grateful :)

**Comment by sebastianhaberey** [comment]
Yes I feel you, I'm also pretty much going by project requirements. Still, I can't promise anything, but if I come up with something for the macOS side of things, I'll get back to you.

**Comment by sebastianhaberey** [comment]
So here's a PR where I updated the project for macOS and then basically copied the iOS Swift code. The NSD API seems to be similar (equivalent?) on both platforms. I was able to detect my printer, so the basics seem to be working.

**Comment by jnstahl** [comment]
Hi,

I am using your plugin on android/ios (thank you very much) but I also need the windows / macos part. Can contribute to windows. I am pretty versed in flutter/android/windows but relatively new to these plugins.

**Comment by Nimrodda** [comment]
@jnstahl macos support is already available but not yet released. It'd be great if you can contribute the Windows implementation.

**Comment by jnstahl** [comment]
@Nimrodda OK on it. I already got it working in principle (my own flutter app is now running on windows and can discover and connect to the server); some caveats:

- for UWP apps there is an actual windows API, but for now I am doing the WIN32 implementation
- in WIN32 there is no API for MDNS, so I had to resort to a public domain solution based on raw sockets
- The platform documentation for the WIN32 plugins is fairly sparse, it took a bit of trial and error to make it work
- The biggest issue right now is that the MDNS code is essentially blocking the main thread while it waits for a response which is not acceptable. the next step is to make it either multithreaded or non-blocking. That might be some heavy lifting. I keep you posted.
